### PR TITLE
Add explicit System.Text.Json dependency

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Contributes to: https://github.com/dotnet/source-build/issues/3540

In source-build, `Microsoft.Build` package dependency is pulling a transitive `System.Text.Json` dependency from PSB. The fix is to simply add an explicit `System.Text.Json` dependency.

I'm not 100% sure if changes in `eng/Version.Details.xml` are necessary.

### CODE REVIEW

**Comment by NikolaMilosavljevic** [comment]
@mthalman @MichaelSimons @mmitche

**Comment by NikolaMilosavljevic** [comment]
@mmitche @JoeRobich @sharwell what is the correct branch that would flow into sdk's `6.0.1xx`? I picked `release/6.0.2xx` but there is also the `release/6.x` branch.

**Comment by JoeRobich** [comment]
Whoops, seems like we missed a branch. I've added a `release/6.0.1xx` branch with appropriate subscriptions which publishes to the expected channel. @sharwell I opened https://github.com/dotnet/roslyn-tools/pull/1355 to update the code flow so update flow appropriately.

**Review by JoeRobich — APPROVED** [review]

File: src/dotnet-format.csproj
Comment (MichaelSimons) [inline src/dotnet-format.csproj:62]: Is it necessary to make this source-build conditioned?

Comment (NikolaMilosavljevic) [reply src/dotnet-format.csproj:62]: I wanted to limit the impact of this change, in servicing. Should we try to make it applicable for all scenarios?

Comment (MichaelSimons) [reply src/dotnet-format.csproj:62]: I'll let the repo owners decide.  My general recommendation is to avoid source-build specific conditions as much as possible.

Comment (NikolaMilosavljevic) [reply src/dotnet-format.csproj:62]: > I'll let the repo owners decide. My general recommendation is to avoid source-build specific conditions as much as possible.

Thanks. @JoeRobich any concerns with making this change applicable for all scenarios and not just source-build?

@MichaelSimons , I have the same condition in the `sdk` change: https://github.com/dotnet/sdk/pull/33618

Comment (NikolaMilosavljevic) [reply src/dotnet-format.csproj:62]: @sharwell do you have any concerns with updating this PR to remove the source-build condition and make the new package reference applicable in all build environments?

Comment (NikolaMilosavljevic) [reply src/dotnet-format.csproj:62]: It seems that we don't have a confirmation yet that source-build condition should be removed. Will merge by EOD unless I hear otherwise.

@sharwell the question was about modifying this PR to remove source-build condition - any thoughts?

**Pull request merged**
